LibTrixi.jl/src/api_jl.jl

```diff
@@ -46,6 +46,12 @@ function trixi_finalize_simulation_jl(simstate)
         end
     end
 
+    # finalize T8codeMesh explicitly
```

Comment:
Why is this required? That is, why not let the finalizer of the Julia object do it for you once it's released by GC?

Reply:
If it is needed e.g. to give back all the memory sooner, I suggest to leave this as a comment in the code. You can then just resolve this convo

Reply:
This is related to the latest changes in Trixi for t8code, see also https://github.com/trixi-framework/Trixi.jl/pull/1980#discussion_r1830524774

Currently `finalize` has to be called, otherwise an error is thrown. We do not have something like an exit hook feature in libtrixi, so you cannot call it from the libelixir.

Reply:
OK. Do you maybe want to add a comment for this?

Reply:
This is an un-feature of t8code interacting with the Julia GC. See this discussion here: https://github.com/DLR-AMR/t8code/issues/1295.

I'm thinking about a solution for that. But in the meantime finalizing explicitly is the best workaround.

Reply:
Comment added.

Reply:
Thanks a lot!